share/scan/registry.go

```diff
@@ -70,52 +70,79 @@ func IsQuayRegistry(rc *RegClient) bool {
 	return strings.EqualFold(rc.URL[:len(quayRegistryURL)], quayRegistryURL)
 }
 
-func BuildV2ImageInfo(dg string, body []byte) (imageInfo ImageInfo, parsedSchemaVersion int, err error) {
+func (rc *RegClient) copyV2LayerAndHistory(imageInfo *ImageInfo, manV2 *manifestV2.Manifest, ccmi *registry.ManifestInfo) bool {
+	allLayersAreCosignPayloads := true
+
+	// In the history list from container image config spec, only the layer that has no empty_layer flag
+	// has a digest in the manifest layer list.
+	// The following section bring the layer list in imageInfo to the same size as history (cmd)
+	j := len(manV2.Layers) - 1
+	for i := 0; i < len(ccmi.Cmds); i++ {
```

Comment:
it looks like ccmi could be nil when copyV2LayerAndHistory() is called

Reply:
OK